packages/slate-editor/src/extensions/paste-images/lib/withImagesPasting.ts

```diff
@@ -0,0 +1,81 @@
+import type { PrezlyFileInfo } from '@prezly/uploadcare';
+import { toProgressPromise, UPLOADCARE_FILE_DATA_KEY, UploadcareImage } from '@prezly/uploadcare';
+import uploadcare from '@prezly/uploadcare-widget';
+import { noop } from '@technically/lodash';
+import type { Editor } from 'slate';
+
+import { filterDataTransferFiles, isFilesOnlyDataTransfer } from '#lib';
+
+import { createImage, IMAGE_TYPES } from '#extensions/image';
+
+import { insertPlaceholders, PlaceholderNode, PlaceholdersManager } from '../../placeholders';
+
+export interface Parameters {
+    onImagesPasted?: (editor: Editor, images: File[]) => void;
+}
+
+export function withImagesPasting({ onImagesPasted = noop }: Parameters = {}) {
+    return <T extends Editor>(editor: T): T => {
+        const parent = {
+            insertData: editor.insertData,
+        };
+
+        editor.insertData = (dataTransfer: DataTransfer) => {
+            if (!isFilesOnlyDataTransfer(dataTransfer)) {
+                // Handle images, if the pasted content is containing files only.
```

Comment:
This comment seems to be incorrect? As this branch will only run if the data transfer is NOT files only.

Reply:
It is correct. As images are handled below this `if`. But I see why you're confused.

How about this?

> Exit, if the pasted content is not containing files only.

Reply:
31ec5e9066b3c03c5f9935d9f15a3094414f064c

Reply:
I think that's clearer 👍